src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -339,18 +361,26 @@ internal bool TryWrite(Utf8JsonWriter writer, in T value, JsonSerializerOptions
                     JsonConverter jsonConverter = state.Current.InitializeReEntry(type, options);
                     if (jsonConverter != this)
                     {
+                        if (options.ReferenceHandlingStrategy == ReferenceHandlingStrategy.IgnoreCycle &&
+                            jsonConverter.IsValueType)
+                        {
+                            // For boxed value types: push the value before it gets unboxed on TryWriteAsObject.
+                            state.ReferenceResolver.PushReferenceForCycleDetection(value);
+                            ignoreCyclesPopReference = true;
+                        }
+
                         // We found a different converter; forward to that.
-                        return jsonConverter.TryWriteAsObject(writer, value, options, ref state);
+                        bool success2 = jsonConverter.TryWriteAsObject(writer, value, options, ref state);
```

Comment:
A variable named `success` already exists (but is declared below this code) in this context.
https://github.com/dotnet/runtime/blob/148bc0820805c7db4ff91d76f3a07bbecc43f160/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs#L384

Reply:
Consider making this one `success` (and the other `success2`) since this appears first; or using an unassigned local higher in the method which can be shared by both call sites.